### ISSUE TITLE
Antennas don't transmit science in 1.05 v

### ISSUE THREAD

**Issue by Jarraya** [RemoteTechnologiesGroup/RemoteTech#511]
Having an issue with the newest version post udate. My space station won't transmit science data. It says transmitting and done, but the science never goes from the station.

**Comment by cool-acid** [comment]
Do you have Science Alert installed?

**Comment by Jarraya** [comment]
No

Short, from my iPhone

**Comment by alexcrawford** [comment]
Same issue reported [here](http://forum.kerbalspaceprogram.com/threads/83305-1-0-5-RemoteTech-v1-6-9-2015-11-10/page373).

**Comment by mikezm** [comment]
That was it for me. Uninstalled Science Alert and I'm able to transmit again.

**Comment by mscg82** [comment]
I have the same problem. After unistalling Science Alert, science experiment can be trasmitted, but the science stored in mobile processing units can't. The antenna which is responsible for transmitting the data get stuck in the "Uploading data..." state.

**Comment by Jarraya** [comment]
Yes, I have the same issue. What I haven't tried is a new mobile processing
unit. Maybe it's only the existing ones.

**Comment by NatTuck** [comment]
I'm having the problem with new labs as well.

**Comment by Varsi** [comment]
Haven't tried labs myself but can't get orbital surveys done. 
The output first shows:
> [R&D]: No Science Subject found with id survey@Kerbin

And after transmitting the packets there's an error:
>NullReferenceException: Object reference not set to an instance of an object
  at ResearchAndDevelopment.GetScienceValue (Single dataAmount, .ScienceSubject subject, Single xmitScalar) [0x00000] in <filename unknown>:0 
  at ResearchAndDevelopment.SubmitScienceData (Single dataAmount, .ScienceSubject subject, Single xmitScalar, .ProtoVessel source, Boolean reverseEngineered) [0x00000] in <filename unknown>:0 
  at RnDCommsStream.submitStreamData (.ProtoVessel source) [0x00000] in <filename unknown>:0 
  at RnDCommsStream.StreamData (Single dataAmount, .ProtoVessel source) [0x00000] in <filename unknown>:0 
  at RemoteTech.Modules.ModuleRTDataTransmitter+<Transmit>c__Iterator0.MoveNext () [0x00000] in <filename unknown>:0 
 (Filename:  Line: -1)

Similar results with other orbital bodies as well.
No other mods than RT 1.6.9 and hyperedit 1.4.2 (same problem without hyperedit).

Attached the whole output.
[output_log.txt](https://github.com/RemoteTechnologiesGroup/RemoteTech/files/40019/output_log.txt)

**Comment by Varsi** [comment]
Using the dll FancyMouse posted at the [forums](http://forum.kerbalspaceprogram.com/threads/83305-1-0-5-RemoteTech-v1-6-9-2015-11-10?p=2302292&viewfull=1#post2302292), the survey works.
Still gives that first error but at least the survey (and possibly to lab) works.

**Comment by jcarnell** [comment]
FancyMouse's dll "fixes" survey scanner and science transmission, breaks Outer Planets Mod.

**Comment by MG2R** [comment]
Affected by this issue as well. Of note: I've seen a lot of people say that science gets removed from their experiments/MPL but it doesn't arrive... My science just stays put. I can start a transmission from my MPL, but it won't do anything, the science just stays in the MPL.

I have no issue transmitting science from experiments, though. Don't have ScienceAlert installed

**Comment by TrickyVic** [comment]
So FancyMouse made a dll, any instructions on how to actually implement the fix for us common folk? What are we supposed to do with the file or line he's written?

**Comment by MG2R** [comment]
Keep in mind that the FancyMouse DLL isn't a fix for everything. If I understand all the numerous comments correctly, it only fixes the issue of transmitting science from experiments when you're using the AlertScience mod.

On my system, the DLL changed nothing. I'm not using AlertScience and have always been able to transmit science from experiments. The issue I have is transmitting science from the MPL. The FancyMouse DLL doesn't seem to fix that.

At any rate, if you want to test it out, you overwrite the RemoteTech DLL (under GameData/RemoteTech/Plugins) with the one FancyMouse compiled. Be sure to back up the original DLL before overwriting it, though ;)

**Comment by Varsi** [comment]
For me the FancyMouses DLL fixed the issue with survey scanner not sending the survey information. I think there was some other fix for Science Alert if it was preventing transmitting science in other situations.

**Comment by Adohk** [comment]
Doesn't transmit science from labs for me, I'd press send but doesn't finish the transmission. No ScienceAlert installed.

**Comment by zegkljan** [comment]
I am also affected by this bug. I right-click the MPL, then left-click the Transmit Science button, then the science is pretending to be transmitted (the progress messages are displayed, energy is consumed) and after 100% the science is not added and is not removed from MPL.

Here is the log from the moment I click on the Transmit Science button up to the end of transmission.

```
[LOG 16:22:04.311] Sending data to vessel comms. 3 devices to choose from. Will try to pick the best one
[LOG 16:22:04.383] Score for transmitter: 1.275 - Data Rate: 6.667 - Data Cost: 7.500
[LOG 16:22:04.383] Score for transmitter: 1.275 - Data Rate: 6.667 - Data Cost: 7.500
[LOG 16:22:04.383] Score for transmitter: 1.275 - Data Rate: 6.667 - Data Cost: 7.500
[ERR 16:22:04.383] [R&D]: No Science Subject found with id sciencelab@Kerbin

[LOG 16:22:05.263] RemoteTech: Changing RnDCommsStream timeout from 0.3 to 0.3200684
[LOG 16:22:05.263] RemoteTech: [Transmitter]: Uploading Data... (Research results) - 6.67 Mits/sec. Packets to go: 56 - Files to Go: 0
[LOG 16:22:06.013] RemoteTech: [Transmitter]: Uploading Data... (Research results) - 6.67 Mits/sec. Packets to go: 55 - Files to Go: 0
... (shortened)
[LOG 16:22:49.089] RemoteTech: [Transmitter]: Uploading Data... (Research results) - 6.67 Mits/sec. Packets to go: 2 - Files to Go: 0
[LOG 16:22:49.839] RemoteTech: [Transmitter]: Uploading Data... (Research results) - 6.67 Mits/sec. Packets to go: 1 - Files to Go: 0
[LOG 16:22:50.602] RemoteTech: [Transmitter]: Uploading Data... (Research results) - 6.67 Mits/sec. Packets to go: 0 - Files to Go: 0
[EXC 16:22:50.602] NullReferenceException: Object reference not set to an instance of an object
	ResearchAndDevelopment.GetScienceValue (Single dataAmount, .ScienceSubject subject, Single xmitScalar)
	ResearchAndDevelopment.SubmitScienceData (Single dataAmount, .ScienceSubject subject, Single xmitScalar, .ProtoVessel source, Boolean reverseEngineered)
	RnDCommsStream.submitStreamData (.ProtoVessel source)
	RnDCommsStream.StreamData (Single dataAmount, .ProtoVessel source)
	RemoteTech.Modules.ModuleRTDataTransmitter+<Transmit>c__Iterator0.MoveNext ()
```

It was a transmission from a station in low Kerbin orbit via 1 relay satellite. I'm running 64b version on Fedora Workstation 23 and I have the latest version of RT available on CKAN (1.6.9). I do not have Science Alert installed (transmitting from experiments works like a charm).

**Comment by jcarnell** [comment]
Drop the dll into the remote tech folder in game data

**Comment by Varsi** [comment]
The dll replaces the original dll in **\gamedata\RemoteTech\Plugins**. I would also suggest keeping the original dll somewhere in case of more trouble.

**Comment by GiraffenLP** [comment]
Saame Problem here with vers. 1.6.9 after uninstall remote tech it works :D but i want this mod so bad

**Comment by billw2012** [comment]
Same issue with the MPL, science transfer simply doesn't do anything. It says it is transferring up to a 100%, sometimes it even says "done" after that, but no science data changes. Please fix this mod is awesome, and so is MPL!

**Comment by tegester** [comment]
I'm having the issue of having science in my Mobile Processing Lab but it is not transmitting. It says transferring and done but the science stays put. Have not tried the dll yet.

**Comment by billw2012** [comment]
Actually turns out there is a RemoteTechXF version of the dll fix as well: http://forum.kerbalspaceprogram.com/index.php?/topic/97924-105-remotetech-xf-remotetech-v169/&do=findComment&comment=2358140

**Comment by lamont-granquist** [comment]
Just validated that I have **never** installed ScienceAlert, but I'm seeing the bug with RT and the MPL not transmitting data.

Note that there's two similar bugs here, one involving ScienceAlert that seems to block transmitting any science data (?) and one not involving ScienceAlert that only blocks transmitting the science which has accumulated in an MPL.

**Comment by jcarnell** [comment]
The RemoteTech XF dll post earlier in the thread worked well for me and appears to still be compatible with Outer Planets Mod as well.

**Comment by Varsi** [comment]
Definitely two different problems.
Science alert, even without remote tech, will prevent any science transmission (at least in career mode). There is a fix for this at [Kerbal forums](http://forum.kerbalspaceprogram.com/index.php?/topic/69538-104-sciencealert-189-experiment-availability-feedback-july-13/&do=findComment&comment=2276279).  
  
The other problem (with Survey Scanner and MPL not finishing transmission) happens with remote tech. There's a fix mentioned earlier in this thread (for remote tech and remote tech XF separately).

Both are very easy to reproduce and don't require any other mods installed.

**Comment by KSPjohnFX** [comment]
Just to add a data point, I currently cannot send data back from a Scansat orbital mapping although no problems so far with other data from small science parts/various antenna (not tried MPL)
(career RSS/RO/RP-0)

**Comment by d4rksh4de** [comment]
Verified in KSP 1.1, MPL science transmission isn't working, it goes to 100%. No science is removed from MPL and nothing is added to the available science. The lab gets into a state where it is not possible to retransmit, says that it is already transmitting, reloading vessel restores the original state.

**Comment by lkucera** [comment]
Seems that this problem is still case in 1.1.2 - I have latest version of RT as part of RSS/RP-0/RO and if I transmit data (for example from Venus orbit) it will transmit say it is 100% and Done! but no science is added...

**Comment by AnghusMcleod** [comment]
So it seems the issue (#667) is still occurring in 1.1.3, at least in relation to SCANsat 16.3 with RSS/RO/RP-0. Satellite with SAR attached in orbit around Moon; tried with both the Commutron32 and LK-TV that I have attached; both same result - says "Done!" and stops there. 

Science is not lost, and re-transmitting a few times in a row doesn't result in success or loss.

If anyone knows a workaround, so I can at least get my science in the meantime...

Log attached:

[KSP_log.txt](https://github.com/RemoteTechnologiesGroup/RemoteTech/files/377156/KSP_log.txt)

**Comment by neitsa** [comment]
stock bug also fixed in 17b2b85 ; see #667 for an explanation.
